src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentService.kt

```diff
@@ -201,5 +208,34 @@ class CodyAgentService(project: Project) : Disposable {
         false
       }
     }
+
+    suspend fun <T> coWithAgent(project: Project, callback: suspend (CodyAgent) -> T) =
+        coWithAgent(project, false, callback)
+
+    suspend fun <T> coWithAgent(
+        project: Project,
+        restartIfNeeded: Boolean,
+        callback: suspend (CodyAgent) -> T
+    ): T {
+      if (!CodyApplicationSettings.instance.isCodyEnabled) {
```

Comment:
That change will require testing all the features with disabled cody:
* chat, @-files 
* commands
* autocompletion
* enhanced context
* edit tasks
* account management panel

I do not trust we properly handle exeptions everywhere.

Reply:
It won't because the `withAgent` wrappers check this boolean. We might get one exception if you disable agent between the time of that check and the time this coroutine runs.